### DISCUSSION TITLE
Bind to send CTRL_T result without confirming?

### DISCUSSION THREAD

**Discussion by indeedwatson** [junegunn/fzf#3780] [General]
Currently pressing enter on a result on ctrl+t puts the path into the terminal and you have to press enter again to send the command.

This is reasonable because you could be selecting multiple paths. 

But in cases where you're not, is there a keybind (or a way to set one) to just send it straight up? The same way alt+c does.

To be clear I'm not looking to replace the behavior, but to enhance it. The bind could be something like shift+enter.

**Comment by LangLangBart** [comment]
Assuming you are using `zsh`.

One solution could be to modify the `fzf-file-widget` function to check if a temporary file is empty. The function will place the file path or execute the command based on this condition.

https://github.com/junegunn/fzf/blob/bcda25a513abc2d4744bc1f8c910eaae305a5242/shell/key-bindings.zsh#L68-L73

```sh
# .zshrc
eval "$(fzf --zsh)"

modified-fzf-file-widget() {
  local temporary_file="/tmp/file_ctrl_t_state"
  LBUFFER="${LBUFFER}$(__fzf_select)"
  if [[ -s $temporary_file ]]; then
    zle accept-line
    : > $temporary_file # clear the file
  fi
  local ret=$?
  zle reset-prompt
  return $ret
}
zle -N modified-fzf-file-widget
bindkey '^T' modified-fzf-file-widget
```

**EDIT:** Added the missing `zle -N modified-fzf-file-widget` line to the code block above.

> The bind could be something like shift+enter.


The key combination <kbd>⇧ Shift</kbd> + <kbd>⏎ Enter</kbd> is not supported.

```sh
man fzf | less --pattern "AVAILABLE KEYS"
```

Let's try using <kbd>⌃ Control</kbd> + <kbd>Space</kbd> for the demo.

```sh
# .zshrc
export FZF_CTRL_T_OPTS="--bind 'ctrl-space:execute-silent(echo 1 >/tmp/file_ctrl_t_state)+accept'"
```

Pressing <kbd>⏎ Enter</kbd> will place the file path in the command line buffer as usual, and <kbd>⌃ Control</kbd> + <kbd>Space</kbd> will execute the entire command.

---

Unsatisfied with the solution, the maintainer used the `--expect` flag for a more elegant approach (no need for temporary file), as demonstrated with the `fzf-history-widget` in this issue: [junegunn/fzf#477](https://github.com/junegunn/fzf/issues/477#issuecomment-230338992).

**Reply by indeedwatson** [reply]
Thanks, I actually use ctrl+space for a similar workflow for zsh's autocomplete, I'll test if I can have both at once because it would be ideal

I'm confused about the last bit of this comment, is that an edit made by junegunn? I'm not too familiar with discussions.

**Reply by LangLangBart** [reply]
> I'm confused about the last bit of this comment, is that an edit made by junegunn?

I don't think the maintainer can edit posts of users on Github discussions.

The comment was just indicating that another way is possible without using a temporary file.
